bats/tests/preferences/verify-paths.bats

```diff
@@ -0,0 +1,114 @@
+# Test case 30
+
+setup() {
+    load '../helpers/load'
+    if is_windows; then
+        skip "test not applicable on Windows"
+    fi
+    # Ensure subshells don't inherit a path that includes ~/.rd/bin
+    PATH_WITHOUT_RD=$(echo "$PATH" | tr ':' '\n' | grep -v /.rd/bin | tr '\n' ':')
+}
+
+teardown_file() {
+    load '../helpers/load'
+    run rdctl shutdown
+    assert_nothing
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start app and fix dotfiles' {
+    start_container_engine
+    wait_for_container_engine
+    ensure_dotfiles_are_completed_BUG_BUG_BUG_4519
+}
+
+# Running `bash -l -c` causes bats to hang
+@test 'bash managed' {
+    run bash -l -c "which rdctl" 3>&-
+    assert_output --partial '.rd/bin/rdctl'
+}
+
+@test 'ksh managed' {
+    if command -v ksh >/dev/null && [ -f "$HOME/.kshrc" ]; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run ksh -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'testing ksh managed: ksh not found'
+    fi
+}
+
+@test 'zsh managed' {
+    if command -v zsh >/dev/null && [ -f "$HOME/.zshrc" ]; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run zsh -i -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'testing zsh managed: zsh not found'
+    fi
+}
+
+@test 'fish managed' {
+    if command -v fish >/dev/null; then
+        PATH="$PATH_WITHOUT_RD"
+        export PATH="$PATH_WITHOUT_RD"
+        run fish -c "which rdctl"
+        assert_success
+        assert_output --partial '.rd/bin/rdctl'
+    else
+        skip 'testing fish managed: fish not found'
+    fi
+}
+
+@test 'move to manual path-management' {
+    rdctl set --application.path-management-strategy=manual
+    sleep 5
```

Comment:
Needing to `sleep` is a bit of a red flag that we have a race condition.

But maybe we'll postpone addressing this to later. We need some mechanism to say "rdctl: wait until all requested changes have been applied".

Reply:
Running `rdctl set...` is async, and chances are good it's going to return before the requested changes have been written to disk.